raiden-dapp/src/components/UploadStateDialog.vue

```diff
@@ -0,0 +1,212 @@
+            }"
+            size="90px"
+          >
+            mdi-upload
+          </v-icon>
+        </v-row>
+        <v-row
+          class="upload-state__dropzone__description"
+          justify="center"
+          no-gutters
+        >
+          {{ $t('backup-state.upload-drag-and-drop') }}
+        </v-row>
+        <v-row
+          class="upload-state__dropzone__description"
+          justify="center"
+          no-gutters
+        >
+          {{ $t('backup-state.upload-divider') }}
+        </v-row>
+        <v-row class="upload-state__dropzone__button">
+          <input ref="stateInput" type="file" hidden @change="onFileSelect" />
+          <action-button
+            :enabled="!activeDropzone ? true : false"
+            ghost
+            :text="$t('backup-state.upload-button')"
+            @click="$refs.stateInput.click()"
+          />
+        </v-row>
+      </div>
+    </v-card-actions>
+  </raiden-dialog>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Emit, Vue } from 'vue-property-decorator';
+import RaidenDialog from '@/components/RaidenDialog.vue';
+import ActionButton from '@/components/ActionButton.vue';
+
+@Component({
+  components: {
+    RaidenDialog,
+    ActionButton
+  }
+})
+export default class UploadStateDialog extends Vue {
+  dragCount: number = 0;
+  activeDropzone: boolean = false;
+  dropzoneErrorMessage: boolean = false;
+  uploadingStateProgress: boolean = false;
+
+  @Prop({ required: true, type: Boolean, default: false })
+  visible!: boolean;
+
+  @Emit()
+  cancel() {}
+
+  onDropzoneEnter(e: DragEvent) {
+    e.preventDefault();
+    this.dragCount++;
+    this.activeDropzone = true;
+  }
+
+  onDropzoneLeave(e: DragEvent) {
+    e.preventDefault();
+    this.dragCount--;
+
+    if (this.dragCount <= 0) {
+      this.activeDropzone = false;
+    }
+  }
+
+  onDropzoneDrop(e: DragEvent) {
+    e.preventDefault();
+    this.activeDropzone = false;
+
+    if (!e.dataTransfer?.files) {
+      this.dropzoneError();
+    }
+
+    const uploadedFile = e.dataTransfer?.files;
+    this.uploadState(uploadedFile!);
+  }
+
+  onFileSelect(e: Event) {
+    if (!(e.target as HTMLInputElement).files) {
+      this.dropzoneError();
+    }
+
+    const uploadedFile = (e.target as HTMLInputElement).files;
+    this.uploadState(uploadedFile!);
+  }
+
+  dropzoneError() {
+    this.uploadingStateProgress = false;
+    this.dropzoneErrorMessage = true;
+    setTimeout(() => {
+      this.dropzoneErrorMessage = false;
+    }, 2000);
+  }
+
+  uploadState(uploadedFile: FileList) {
+    if (uploadedFile.length > 1) {
+      this.dropzoneError();
+    }
+
+    let reader = new FileReader();
+    /* istanbul ignore next */
+    reader.onload = e => {
+      if (!e.target?.result) {
+        this.dropzoneError();
+      }
+
+      try {
+        this.uploadingStateProgress = true;
+        const retrievedState = e.target?.result;
```

Comment:
I would probably extract the `e.target` to a const at the beginning of the `onload`, check if it is set there and return if it is not. This would stop the execution from proceeding if the target is falsy and will remove the repetition of checking for nullability